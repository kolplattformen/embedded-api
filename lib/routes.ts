export const login = (personalNumber: string) =>
  `https://login003.stockholm.se/NECSadcmbid/authenticate/NECSadcmbid?TARGET=-SM-HTTPS%3a%2f%2flogin001%2estockholm%2ese%2fNECSadc%2fmbid%2fb64startpage%2ejsp%3fstartpage%3daHR0cHM6Ly9ldGphbnN0LnN0b2NraG9sbS5zZS92YXJkbmFkc2hhdmFyZS9pbmxvZ2dhZDIvaGVt&initialize=bankid&personalNumber=${personalNumber}&_=${Date.now()}`

export const loginStatus = (order: string) =>
  `https://login003.stockholm.se/NECSadcmbid/authenticate/NECSadcmbid?TARGET=-SM-HTTPS%3a%2f%2flogin001%2estockholm%2ese%2fNECSadc%2fmbid%2fb64startpage%2ejsp%3fstartpage%3daHR0cHM6Ly9ldGphbnN0LnN0b2NraG9sbS5zZS92YXJkbmFkc2hhdmFyZS9pbmxvZ2dhZDIvaGVt&verifyorder=${order}&_=${Date.now()}`

export const loginCookie =
  'https://login003.stockholm.se/NECSadcmbid/authenticate/SiteMinderAuthADC?TYPE=33554433&REALMOID=06-42f40edd-0c5b-4dbc-b714-1be1e907f2de&GUID=&SMAUTHREASON=0&METHOD=GET&SMAGENTNAME=IfNE0iMOtzq2TcxFADHylR6rkmFtwzoxRKh5nRMO9NBqIxHrc38jFyt56FASdxk1&TARGET=-SM-HTTPS%3a%2f%2flogin001%2estockholm%2ese%2fNECSadc%2fmbid%2fb64startpage%2ejsp%3fstartpage%3daHR0cHM6Ly9ldGphbnN0LnN0b2NraG9sbS5zZS92YXJkbmFkc2hhdmFyZS9pbmxvZ2dhZDIvR2V0Q2hpbGRyZW4%3d'

const urlLoggedIn = `https://etjanst.stockholm.se/vardnadshavare/inloggad2`

export const children = `${urlLoggedIn}/GetChildren`

export const calendar = (childId: string) =>
  `${urlLoggedIn}/Calender/GetSchoolCalender?childId=${childId}&rowLimit=50`

export const classmates = (childId: string) =>
  `${urlLoggedIn}/contacts/GetStudentsByClass?studentId=${childId}`

export const user =
  'https://etjanst.stockholm.se/vardnadshavare/base/getuserdata'

export const news = (childId: string) =>
  `${urlLoggedIn}/News/GetNewsArchive?bannerImageLimit=5000&childId=${childId}`

export const newsDetails = (childId: string, newsId: string) =>
  `${urlLoggedIn}/News/GetNewsArticle?newsItemId=${newsId}&childId=${childId}`

export const image = (url: string) => `${urlLoggedIn}/NewsBanner?url=${url}`

export const notifications = (childId: string) =>
  `${urlLoggedIn}/Overview/GetNotification?childId=${childId}`

export const menuRss = (childId: string) =>
  `${urlLoggedIn}/Matsedel/GetMatsedelRSS?childId=${childId}`

export const menuList = (childId: string) =>
  `${urlLoggedIn}/Matsedel/GetMatsedelList?childId=${childId}`

export const menuChoice = (childId: string) =>
  `${urlLoggedIn}/Matsedel/GetMatsedelChoice?childId=${childId}`

export const schedule = (childId: string, fromDate: string, endDate: string) =>
  `${urlLoggedIn}/Calender/GetSchema?childId=${childId}&startDate=${fromDate}&endDate=${endDate}`

export const cdn = 'https://etjanst.stockholm.se/vardnadshavare/base/cdn'

export const auth = 'https://etjanst.stockholm.se/vardnadshavare/base/auth'

export const startBundle =
  'https://etjanst.stockholm.se/vardnadshavare/bundles/start'

export const hemPage =
  'https://etjanst.stockholm.se/vardnadshavare/inloggad2/hem'

export const navigationControllerScript =
  'https://etjanst.stockholm.se/vardnadshavare/bundles/navigationController'

export const baseEtjanst = 'https://etjanst.stockholm.se'

export const childcontrollerScript = `https://etjanst.stockholm.se/vardnadshavare/bundles/childcontroller?v=${Date.now()}`

export const createItemConfig =
  'https://raw.githubusercontent.com/kolplattformen/embedded-api/main/config.json'
