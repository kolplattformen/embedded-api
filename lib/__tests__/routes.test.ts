import * as routes from '../routes'

test.each([
  ['children', routes.children],
  ['calender', routes.calendar('123')],
  ['classmates', routes.classmates('123')],
  ['classmates', routes.user],
  ['news', routes.news('123')],
  ['newsDetails', routes.newsDetails('123')],
  ['image', routes.image('123')],
  ['notifications', routes.notifications('123')],
  ['menuRss', routes.menuRss('123')],
  ['menuList', routes.menuList('123')],
  ['menuChoice', routes.menuChoice('123')],
  ['schedule', routes.schedule('123')],
])('handles route %s', (_name, input) => {
  expect(input).toMatchSnapshot()
})
