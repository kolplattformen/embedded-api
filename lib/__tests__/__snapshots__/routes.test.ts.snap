// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles route calender 1`] = `"https://etjanst.stockholm.se/vardnadshavare/inloggad2/Calender/GetSchoolCalender?childId=123&rowLimit=50"`;

exports[`handles route children 1`] = `"https://etjanst.stockholm.se/vardnadshavare/inloggad2/GetChildren"`;

exports[`handles route classmates 1`] = `"https://etjanst.stockholm.se/vardnadshavare/inloggad2/contacts/GetStudentsByClass?studentId=123"`;

exports[`handles route classmates 2`] = `"https://etjanst.stockholm.se/vardnadshavare/base/getuserdata"`;

exports[`handles route image 1`] = `"https://etjanst.stockholm.se/vardnadshavare/inloggad2/NewsBanner?url=https://example.com/img.png"`;

exports[`handles route menuChoice 1`] = `"https://etjanst.stockholm.se/vardnadshavare/inloggad2/Matsedel/GetMatsedelChoice?childId=123"`;

exports[`handles route menuList 1`] = `"https://etjanst.stockholm.se/vardnadshavare/inloggad2/Matsedel/GetMatsedelList?childId=123"`;

exports[`handles route menuRss 1`] = `"https://etjanst.stockholm.se/vardnadshavare/inloggad2/Matsedel/GetMatsedelRSS?childId=123"`;

exports[`handles route news 1`] = `"https://etjanst.stockholm.se/vardnadshavare/inloggad2/News/GetNewsArchive?bannerImageLimit=5000&childId=123"`;

exports[`handles route newsDetails 1`] = `"https://etjanst.stockholm.se/vardnadshavare/inloggad2/News/GetNewsArticle?newsItemId=321&childId=123"`;

exports[`handles route notifications 1`] = `"https://etjanst.stockholm.se/vardnadshavare/inloggad2/Overview/GetNotification?childId=123"`;

exports[`handles route schedule 1`] = `"https://etjanst.stockholm.se/vardnadshavare/inloggad2/Calender/GetSchema?childId=123&startDate=2021-01-01&endDate=2021-01-01"`;
